### PR TITLE
Add collection schema support to CLI

### PR DESCRIPTION
**Pull request by MHove89**
Collection schema support has been added to the CLI

### CODE REVIEW

File: src/Enterspeed.Cli/Services/FileService/SchemaFileService.cs [hunk 1]
```diff
@@ -148,7 +160,7 @@ private void EnsureSchemaFolders(string schemaName = null)
             {
                 var schemasDirectoryName = _schemaNameService.GetSchemasDirectoryName();
                 var schemaDirectoryPath = _filePathService.GetDirectoryPathBySchemaName(schemaName, schemasDirectoryName);
-                if (!Directory.Exists(schemaDirectoryPath))
+                if (!Directory.Exists(schemaDirectoryPath) && !string.IsNullOrEmpty(schemaDirectoryPath))
```
Comment (MHove89) [inline src/Enterspeed.Cli/Services/FileService/SchemaFileService.cs:163]: Getting rid of possible null reference warning

File: src/Enterspeed.Cli/Services/FileService/SchemaFileService.cs [hunk 2]
```diff
@@ -216,6 +228,11 @@ private SchemaType GetSchemaTypeFromFilePath(string filePath)
             return SchemaType.Partial;
         }
 
+        if (Regex.IsMatch(filePath, ".*.collection.(?:js)$"))
```
[the remark this answers is not in the document]
Comment (MHove89) [reply src/Enterspeed.Cli/Services/FileService/SchemaFileService.cs]: This has been implemented

File: src/Enterspeed.Cli/Services/FileService/SchemaFileService.cs [hunk 3]
```diff
@@ -70,7 +74,15 @@ private void CreateJavascriptSchemaFile(SchemaType schemaType, MappingSchemaVers
         // Version does not have any data. Assign default js setup for js schemas instead as a temp fix. 
         if (schemaVersion.Data == null)
         {
-            var byteArray = Encoding.UTF8.GetBytes(schemaType == SchemaType.Partial ? DefaultJsPartialContent : DefaultJsContent);
+            var defaultJsContent = schemaType switch
+            {
+                SchemaType.Normal => DefaultJsContent,
```
[the remark this answers is not in the document]
Comment (MHove89) [reply src/Enterspeed.Cli/Services/FileService/SchemaFileService.cs]: This has been implemented

File: src/Enterspeed.Cli/Services/FileService/SchemaFileService.cs [hunk 4]
```diff
@@ -216,23 +216,24 @@ private string GetFilePath(string schemaName, string alias, SchemaType schemaTyp
...
             return SchemaType.Normal;
         }
+
         if (Regex.IsMatch(filePath, ".*.partial.(?:js|json)$"))
         {
             return SchemaType.Partial;
         }
 
-        if (Regex.IsMatch(filePath, ".*.collection.(?:js)$"))
+        if (Regex.IsMatch(filePath, ".*.partial.(?:js|json)$"))
```
Comment (jesperweber) [inline src/Enterspeed.Cli/Services/FileService/SchemaFileService.cs:232]: Is this a copy paste mistake, I guess it should be collection and not partial?

Comment (MHove89) [reply src/Enterspeed.Cli/Services/FileService/SchemaFileService.cs]: This has now been fixed

**Review by jesperweber — APPROVED** [review]

**Pull request merged**